### PR TITLE
Added OSX and Ubuntu 16.04 to the pr job builders.

### PR DESCRIPTION
**Pull request by 333fred**
This adds OSX to the list of PR jobs so that we verify all commits on it.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet-bot test ci please

**Review by nguerrera — APPROVED** [review]

**Comment by nguerrera** [comment]
Retarget to master after #917 goes in

**Comment by 333fred** [comment]
@dotnet-bot test ci please

**Review by nguerrera — APPROVED** [review]

**Review by livarcocc — APPROVED** [review]

**Comment by dsplaisted** [comment]
Are we waiting on anything to merge this? @333fred @nguerrera

**Comment by 333fred** [comment]
It still doesn't work. See the offline mail thread.

**Review by nguerrera — CHANGES_REQUESTED** [review]
Preventing merge based on the fact that this isn't working (see above)

**Comment by 333fred** [comment]
@dotnet-bot test ci please

**Comment by eerhardt** [comment]
What's the status of this? We've introduced more "OSX" specific failures in the meantime: See #965

**Comment by 333fred** [comment]
@dotnet-bot test ci please

**Comment by 333fred** [comment]
Setting HOME appears to have made the build successful. @nguerrera, can you approve?

**Comment by 333fred** [comment]
Test failures on OSX, appear to be the same ones in #965.

File: build.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ while [[ $# > 0 ]]; do
             args=( "${args[@]/$1}" )
             args=( "${args[@]/$2}" )
             shift
-            ;;        
+            ;;
```
Comment (nguerrera) [inline build.sh]: @333fred I keep seeing unrelated whitespace cleanup. Is your editor configured to do that? It's not code review friendly.

**Comment by eerhardt** [comment]
> Test failures on OSX, appear to be the same ones in #965.

Working on fixing them now.

File: build.sh [hunk 2]
```diff
@@ -68,6 +68,14 @@ export NUGET_HTTP_CACHE_PATH="$REPOROOT/packages"
 [ -z "$DOTNET_INSTALL_DIR" ] && export DOTNET_INSTALL_DIR=$REPOROOT/.dotnet_cli
 [ -d "$DOTNET_INSTALL_DIR" ] || mkdir -p $DOTNET_INSTALL_DIR
 
+# Some things depend on HOME and it may not be set. We should fix those things, but until then, we just patch a value in
+if [ -z "$HOME" ]; then
+    export HOME="$REPOROOT/.home"
+
+    [ ! -d "$HOME" ] || rm -Rf $HOME
```
Comment (nguerrera) [inline build.sh]: This line is terrifying. Do we really need to delete it? CI does a clean of the enlistment, right? And we don't put stuff in there.

File: build.sh [hunk 3]
```diff
@@ -55,18 +55,27 @@ done
...
 
 # Set nuget package cache under the repo
 export NUGET_PACKAGES="$REPOROOT/packages"
+export NUGET_HTTP_CACHE_PATH="$REPOROOT/packages"
 
 # Use a repo-local install directory (but not the artifacts directory because that gets cleaned a lot
 [ -z "$DOTNET_INSTALL_DIR" ] && export DOTNET_INSTALL_DIR=$REPOROOT/.dotnet_cli
 [ -d "$DOTNET_INSTALL_DIR" ] || mkdir -p $DOTNET_INSTALL_DIR
 
+# Some things depend on HOME and it may not be set. We should fix those things, but until then, we just patch a value in
```
Comment (nguerrera) [inline build.sh]: This should say what things and link to bugs.

File: netci.groovy [hunk 4]
```diff
@@ -10,7 +10,7 @@ def project = GithubProject
 def branch = GithubBranchName
 def isPR = true
 
-def osList = ['Windows_NT', 'Windows_NT_FullFramework', 'Ubuntu14.04']
+def osList = ['Windows_NT', 'Windows_NT_FullFramework', 'Ubuntu14.04', 'Ubuntu16.04', 'OSX']
```
Comment (nguerrera) [inline netci.groovy]: Does this include 10.12?

Comment (333fred) [inline netci.groovy]: Not sure, @mmitche? What OS is the OSX host?

File: build.sh (hunk 2 above)
Comment (333fred) [inline build.sh]: I copied it from the CliCommandLineParser repo. @jonsequitur, was there a reasoning behind deleting everything?

File: netci.groovy (hunk 4 above)
Comment (eerhardt) [inline netci.groovy]: See https://github.com/dotnet/corefx/pull/16562/files for how corefx explicitly includes 10.12

File: build.sh (hunk 1 above)
Comment (333fred) [inline build.sh]: Yes, my editor trims whitespace. Which works fine when I'm creating a file, but less fine when I'm not.

File: build.sh [hunk 5]
```diff
@@ -68,7 +68,7 @@ export NUGET_HTTP_CACHE_PATH="$REPOROOT/packages"
 [ -z "$DOTNET_INSTALL_DIR" ] && export DOTNET_INSTALL_DIR=$REPOROOT/.dotnet_cli
 [ -d "$DOTNET_INSTALL_DIR" ] || mkdir -p $DOTNET_INSTALL_DIR
 
-# Some things depend on HOME and it may not be set. We should fix those things, but until then, we just patch a value in
+# NuGet depends on HOME and it may not be set. Until it's fixed, we just patch a value in
```
Comment (nguerrera) [inline build.sh]: Bug filed?

**Review by nguerrera — APPROVED** [review]

File: build.sh (hunk 2 above)
Comment (eerhardt) [inline build.sh]: I have an extremely strong feeling that CliCommandLineParser just copied it from

https://github.com/dotnet/cli/blob/master/build.sh#L19-L25

which came in over a year ago here:
https://github.com/dotnet/cli/commit/34b0b688f7db70003514e0aefbee47868e6d5f71
with the commit message `patch up some final missing CI build things`

**Comment by eerhardt** [comment]
test ci please

**Comment by eerhardt** [comment]
@333fred - I'm getting errors on the OSX run with your latest changes:

https://ci.dot.net/job/dotnet_sdk/job/master/job/GenPRTest/job/release_osx10.12_prtest/1/console

```
13:09:16 Unhandled Exception: System.ArgumentNullException: Value cannot be null.
13:09:16 Parameter name: path1
13:09:16    at System.IO.Path.Combine(String path1, String path2, String path3)
13:09:16    at NuGet.Common.NuGetEnvironment.GetFolderPath(SpecialFolder folder)
13:09:16    at NuGet.Common.NuGetEnvironment.GetFolderPath(NuGetFolderPath folder)
13:09:16    at NuGet.Configuration.SettingsUtility.GetHttpCacheFolder()
13:09:16    at NuGet.Configuration.NuGetPathContext.Create(ISettings settings)
13:09:16    at Microsoft.DotNet.Configurer.NuGetCacheSentinel.get_NuGetCachePath()
13:09:16    at Microsoft.DotNet.Configurer.NuGetCacheSentinel.Exists()
13:09:16    at Microsoft.DotNet.Cli.Telemetry..ctor(INuGetCacheSentinel sentinel, String sessionId)
13:09:16    at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
13:09:16    at Microsoft.DotNet.Cli.Program.Main(String[] args)
```

are the NuGet env vars set up correctly?

**Comment by eerhardt** [comment]
@333fred @nguerrera - OSX.10.12 now passes:

https://ci.dot.net/job/dotnet_sdk/job/master/job/GenPRTest/job/release_osx10.12_prtest/3/

I'll let you guys merge when ready.

**Pull request merged**